google/auth/credentials.py

```diff
@@ -18,17 +18,21 @@
 import abc
 from enum import Enum
 import os
+from typing import Optional, Self, Sequence
 
 from google.auth import _helpers, environment_vars
 from google.auth import exceptions
 from google.auth import metrics
-from google.auth._credentials_base import _BaseCredentials
+from google.auth._credentials_base import BaseCredentials
 from google.auth._refresh_worker import RefreshThreadManager
+from google.auth.credentials import Credentials
```

Comment:
We're importing from the file itself? This doesn't seem right.

Reply:
```suggestion
```

Reply:
Ah, yeah, sorry you're right